Feature/fluxc medialib recycler

nbradbury opened the pull request:
Rewrites the media browser & picker to use a RecyclerView, RecyclerView.Adapter, and GridLayoutManager. This replaces the outdated GridView, which was the source of some problems (most notably missing/incorrect images when scrolling).

Note that there are still some media browser/picker issues & changes as per #5337, but these will be addressed separately.

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryPickerActivity.java:
```diff
@@ -97,24 +96,29 @@ protected void onCreate(Bundle savedInstanceState) {
-        mGridAdapter = new MediaGridAdapter(this, mSite, null, 0, WordPress.sImageLoader);
-        mGridAdapter.setSelectedItems(selectedItems);
+        mGridAdapter = new MediaGridAdapter(this, mSite, WordPress.sImageLoader);
         mGridAdapter.setCallback(this);
```
while not part of the PR in a direct way, I had to double check whether this could lead to potential leaks. I think we're safe, but asking you too to make sure: this would be a circular reference, right? so the Java GC would take care of this as the Activity wouldn't be chained in case it needs be destroyed. Can you confirm / double check please?

nbradbury replied:
Passing the activity as the callback to an adapter is fairly common practice, and I don't think there's a risk of a leak (but good you pointed it out so I could refresh my memory about it!).

mzorz replied:
Awesome, thank you Nick!

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryPickerActivity.java:
```diff
@@ -190,67 +192,55 @@ public void onMediaListFetched(OnMediaListFetched event) {
+            Intent intent = new Intent();
+            int localId = mGridAdapter.getLocalMediaIdAtPosition(position);
+            ArrayList<Long> remoteMediaIds = new ArrayList<>();
+            remoteMediaIds.add(mMediaStore.getMediaWithLocalId(localId).getMediaId());
```
would it be possible that `getMediaWithLocalId` returns null here?

mzorz commented on WordPress/src/main/java/org/wordpress/android/ui/media/MediaGalleryPickerActivity.java:
```diff
@@ -279,7 +270,7 @@ private void refreshMediaFromServer(boolean loadMore) {
+        if (mGridAdapter.getSelectedItemCount() > 0) {
             ArrayList<Long> remoteMediaIds = new ArrayList<>();
             for (Integer localId : mGridAdapter.getSelectedItems()) {
                 remoteMediaIds.add(mMediaStore.getMediaWithLocalId(localId).getMediaId());
```
would it be possible that `getMediaWithLocalId` returns null here?

nbradbury commented:
Thanks as always for the review @mzorz! I added the missing null checks in 7beb317, and I also fixed an issue in 5ee6828 where the imageView could show the previous image for non-image media files.

mzorz commented:
:shipit:
